### ISSUE TITLE
Provider not defined

### ISSUE THREAD

**Issue by wmeneses** [edgardmessias/glpi-singlesignon#26]
I have tried with Azure or generic, but in both options I get this option

**Comment by champakakumara** [comment]
configured with google ,got error 
"Provider not defined."

**Comment by astep3000** [comment]
SOLVED
in file provider.class.php
change in line 907:
'redirect_uri' => $this->getCurrentURL(),
to 
'redirect_uri' => 'https://YOUR-SITE-NAME/plugins/singlesignon/front/callback.php/provider/',

**Comment by champakakumara** [comment]
> 

hi astep3000

Please provide me file path provider.class.php and callback.php

**Comment by champakakumara** [comment]
> SOLVED for google in file provider.class.php change in line 907: 'redirect_uri' => $this->getCurrentURL(), to 'redirect_uri' => 'https://YOUR-SITE-NAME/plugins/singlesignon/front/callback.php/provider/',
> 
> and
> 
> in file callback.php in line 12
> 
> $provider_id = 1; //PluginSinglesignonToolbox::getCallbackParameters('provider'); insert ID

HI

I change above lines but got error
" Error 400: redirect_uri_mismatch"
any Idea?

**Comment by astep3000** [comment]
fill in DB table "glpi_plugin_singlesignon_providers" manually:
![image](https://user-images.githubusercontent.com/24912077/169214532-9cca4587-3350-40cf-b328-2421ad15c33c.png)

**Comment by astep3000** [comment]
![image](https://user-images.githubusercontent.com/24912077/169215378-c34ae273-9e8c-435a-b340-3254f589bbd4.png)

**Comment by champakakumara** [comment]
> ![image](https://user-images.githubusercontent.com/24912077/169215378-c34ae273-9e8c-435a-b340-3254f589bbd4.png)

thanks, now work and redirect from google but login failed

![image](https://user-images.githubusercontent.com/104112313/169242692-50a37443-7e26-4201-914e-fc5820c01387.png)

**Comment by champakakumara** [comment]
![image](https://user-images.githubusercontent.com/104112313/169259210-42aedde5-4de3-457e-9882-80d4d1b403a9.png)

![image](https://user-images.githubusercontent.com/104112313/169259443-fcbafa94-dad6-4361-a049-107689a58d46.png)

same error

**Comment by astep3000** [comment]
may be space?
del space
![image](https://user-images.githubusercontent.com/24912077/169260163-df439406-9edc-4b99-b370-1f4f545621e9.png)

**Comment by champakakumara** [comment]
> ![image](https://user-images.githubusercontent.com/104112313/169259210-42aedde5-4de3-457e-9882-80d4d1b403a9.png)
> 
> ![image](https://user-images.githubusercontent.com/104112313/169259443-fcbafa94-dad6-4361-a049-107689a58d46.png)
> 
> email address configured to glpi user and configured popupAuth and authorized Domain but same error

what I misconfigure?

**Comment by champakakumara** [comment]
test with one domain but same result

**Comment by champakakumara** [comment]
> may be space? del space ![image](https://user-images.githubusercontent.com/24912077/169260163-df439406-9edc-4b99-b370-1f4f545621e9.png)
> 
> or try leaving one domain for test

test with one domain but same error

**Comment by wmeneses** [comment]
😄 It's the same case with AZURE, although I only modified the callback file and it worked fine, thanks for enlightening my way.

**Comment by edgardmessias** [comment]
Hi everyone, I have been working on anothers projects at moment, and because of that I couldn't see these kind of problems.

Pull request are welcome

**Comment by edgardmessias** [comment]
Could you update to latest release and run "Test" from config page?

https://github.com/edgardmessias/glpi-singlesignon/releases/tag/v1.3.3

**Comment by champakakumara** [comment]
I installed new version but got error again " Authorization Error ,Error 400: redirect_uri_mismatch"

**Comment by champakakumara** [comment]
> 

Now Work ,I changed  provider.class.php callback URL. now work thanks

**Comment by champakakumara** [comment]
New version work with google SSO

**Comment by champakakumara** [comment]
Another issue only one user can connect (only my email-admin) other user same domain or different domain can't connect. Error "user not authorized to connect toGLPI"

**Comment by sachaw0lf** [comment]
Hi all, 
I follow this issue and the workarounds but I still have the problem : URI Mismatch. : 
AADSTS50011: The redirect URI 'https://xxxxxxxxxx/plugins/singlesignon/front/callback.php/provider/Array' 
As You can see, the provider ID is Array and not 2 like in configuration plugin page.

I modified the files : front/callback.php and inc/provider.class.php but nothing change.
I am with version 1.3.3 
Thanks

**Comment by edgardmessias** [comment]
Hi, I fixed the url callback in the master branch.

**Comment by sachaw0lf** [comment]
I just tested with the fix, it works perfectly.
Thanks a lot !

**Comment by AdminCeci** [comment]
Hello, I have same issue when try to log with Google Account 'Provider not definied'
When i do the test 'Test single sign-on' it works fine.

The call address when try to log with Google is 'plugins/singlesignon/front/callback.php/provider/Array' 
I setup the callback URI with plugins/singlesignon/front/callback.php/provider/1 on Goole API

Any idea ?

**Comment by Megachip** [comment]
> Hello, I have same issue when try to log with Google Account 'Provider not definied' When i do the test 'Test single sign-on' it works fine.
> 
> The call address when try to log with Google is 'plugins/singlesignon/front/callback.php/provider/Array' I setup the callback URI with plugins/singlesignon/front/callback.php/provider/1 on Goole API
> 
> Any idea ?

https://github.com/edgardmessias/glpi-singlesignon/commit/c7b1be17b3c2d80c6df31d334d3c7d3de1214a74

**Comment by RaulGrosmam** [comment]
Hello,

Did someone make it work properly using Keycloak (using Azure as Login method)?

I've done all the troubleshooting of this Issue but I can't pass through "User is not authorized to connect in GLPI"

This GLPI had LDAP Authentication, so it has a lot of legacy users, it may be a problem?

Follow my current configuration:

![image](https://github.com/edgardmessias/glpi-singlesignon/assets/49730433/f0e88a29-5117-430b-9933-821451e7fd7e)
![image](https://github.com/edgardmessias/glpi-singlesignon/assets/49730433/11da067c-6b9f-4a24-af84-0ab70fc8e203)
![image](https://github.com/edgardmessias/glpi-singlesignon/assets/49730433/32a3545c-6394-4d4f-8e3b-82ceb4c0fcf2)

**Comment by Phantom505** [comment]
H
<img width="280" alt="223Capture" src="https://github.com/edgardmessias/glpi-singlesignon/assets/53447422/afa69ff7-0104-42c8-947a-7613a2502647">
ello, I updated the plugin in the system to the latest version, corrected the php files as indicated in this thread, but it still gives an error as in the screenshot. What could be the problem?

glpi+keycloak

**Comment by Phantom505** [comment]
H
<img width="280" alt="223Capture" src="https://github.com/edgardmessias/glpi-singlesignon/assets/53447422/b14e791b-502e-4936-bd22-d83b098aeb97">
ello, I updated the plugin in the system to the latest version, corrected the php files as indicated in this thread, but it still gives an error as in the screenshot. What could be the problem?

glpi+keycloak

**Comment by RaulGrosmam** [comment]
> H <img alt="223Capture" width="280" src="https://user-images.githubusercontent.com/53447422/257435449-b14e791b-502e-4936-bd22-d83b098aeb97.PNG"> ello, I updated the plugin in the system to the latest version, corrected the php files as indicated in this thread, but it still gives an error as in the screenshot. What could be the problem?
> 
> glpi+keycloak

Check if the user was created an if it has any kind of permission to access GLPI

**Comment by Phantom505** [comment]
> > H <img alt="223Capture" width="280" src="https://user-images.githubusercontent.com/53447422/257435449-b14e791b-502e-4936-bd22-d83b098aeb97.PNG"> ello, I updated the plugin in the system to the latest version, corrected the php files as indicated in this thread, but it still gives an error as in the screenshot. What could be the problem?
> > glpi+keycloak
> 
> Check if the user was created an if it has any kind of permission to access GLPI

a user has been created on the keycloak side, there are accesses between servers, a button appears in glpi through which I pass authorization on keycloak, then it redirects me to glpi where I get this error

**Comment by SkoEdi** [comment]
i just try to setup the plugin with Azure and i had the error Provider not defined.
I juste change in callback.php in line 12

$provider_id = PluginSinglesignonToolbox::getCallbackParameters('provider');

I change it like below

$provider_id =1;

I had to adapt the uri and then it work:
before: www.myglpi.com/plugins/singlesignon/front/callback.php/provider/1
after: www.myglpi.complugins/singlesignon/front/callback.php/provider/Array

**Comment by colimberk** [comment]
> > 
> 
> Ahora trabajo, cambié la URL de devolución de llamada de proveedor.class.php. ahora trabajo gracias

Good afternoon engineer.

I would like to ask you if you have any guide that allows you to correctly install this plugin.
I have 2 problems, 1 is that the user is not validated in AZURE if not BD GLPI, 2 when you press the login button and there is a Microsoft session open, it tries to directly enter the system without asking for credentials. Thank you so much.
Excuse the translation.

**Comment by champakakumara** [comment]
![image](https://github.com/edgardmessias/glpi-singlesignon/assets/104112313/d56b1e03-c500-4b2f-a06b-a2b296545cf3)

**Comment by champakakumara** [comment]
![image](https://github.com/edgardmessias/glpi-singlesignon/assets/104112313/2d29f1d4-3e93-4806-b5d6-18756d616113)

**Comment by muzzol** [comment]
I would like to add some info in case someone else is facing same problem.

Go to Configuration -> General and check public URL

for me that URL was misconfigured (I use a proxypass in front) and the callback was not working properly until I corrected that setting.

I hope this saves you some time.

**Comment by eduardomozart** [comment]
@RaulGrosmam please open a separated issue if your problem persists.

**Comment by eduardomozart** [comment]
@muzzol recently a commit has been made to the plug-in to honor the "Application URL" setup into General > Setup and if it's incorrectly set it will be reflected on "Callback URL" of the plug-in.

**Comment by eduardomozart** [comment]
Hello guys,
Actually there should be no need to modify the PHP function ``getCallbackURL`` as it has been fixed and returns the correct provider ID from provider plug-in settings. Also, the "AuthorizedDomains" tooltip has been updated to reflect the format cited by @astep3000.
Anyone facing issues with the plu-gin please open a new issue request for us to troubleshoot it further.

**Comment by muzzol** [comment]
thanks a lot for your time @eduardomozart 

I see more people getting affected by URL rewrites so your fix is really welcomed.

obrigado!
